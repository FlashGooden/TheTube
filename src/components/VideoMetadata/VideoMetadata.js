import React from "react";
import { Button, Divider, Icon } from "semantic-ui-react";
import "./VideoMetadata.scss";
import { Rating } from "../Rating/Rating";

export function VideoMetadata(props) {
  const viewCount = Number(props.viewCount).toLocaleString() || "";
  /* Todo */
  return (
    <div className="video-metadata" style={props.style}>
      <h3>video title</h3>
      <div className="video-stats">
        <span>{viewCount} views </span>
        <div className="video-actions">
          <Rating likeCount={300} dislikeCount={40} />
          <Button basic icon labelPosition="left">
            <Icon name="share" />
          </Button>
          <Button basic icon>
            <Icon name="add circle" />
          </Button>
          <Button basic icon>
            <Icon name="ellipsis horizontal" />
          </Button>
        </div>
      </div>
      <Divider />
    </div>
  );
}
