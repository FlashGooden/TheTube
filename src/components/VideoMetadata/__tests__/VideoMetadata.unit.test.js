import React from 'react'
import {shallow} from 'enzyme'
import {VideoMetadata} from '../VideoMetadata'

test('Renders VideoMetadata', ()=>{
  const wrapper = shallow(
    <VideoMetadata />
  );
  expect(wrapper).toMatchSnapshot()
})
test('Renders VideoMetadata with viewCount', ()=>{
  const wrapper = shallow(
    <VideoMetadata viewCount={10004}/>
  );
  expect(wrapper).toMatchSnapshot()
})