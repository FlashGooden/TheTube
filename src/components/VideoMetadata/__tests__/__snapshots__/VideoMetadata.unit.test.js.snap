// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders VideoMetadata 1`] = `
<div
  className="video-metadata"
>
  <h3>
    video title
  </h3>
  <div
    className="video-stats"
  >
    <span>
      NaN
       views 
    </span>
    <div
      className="video-actions"
    >
      <Rating
        dislikeCount={40}
        likeCount={300}
      />
      <Button
        as="button"
        basic={true}
        icon={true}
        labelPosition="left"
      >
        <Icon
          as="i"
          name="share"
        />
      </Button>
      <Button
        as="button"
        basic={true}
        icon={true}
      >
        <Icon
          as="i"
          name="add circle"
        />
      </Button>
      <Button
        as="button"
        basic={true}
        icon={true}
      >
        <Icon
          as="i"
          name="ellipsis horizontal"
        />
      </Button>
    </div>
  </div>
  <Divider />
</div>
`;

exports[`Renders VideoMetadata with viewCount 1`] = `
<div
  className="video-metadata"
>
  <h3>
    video title
  </h3>
  <div
    className="video-stats"
  >
    <span>
      10,004
       views 
    </span>
    <div
      className="video-actions"
    >
      <Rating
        dislikeCount={40}
        likeCount={300}
      />
      <Button
        as="button"
        basic={true}
        icon={true}
        labelPosition="left"
      >
        <Icon
          as="i"
          name="share"
        />
      </Button>
      <Button
        as="button"
        basic={true}
        icon={true}
      >
        <Icon
          as="i"
          name="add circle"
        />
      </Button>
      <Button
        as="button"
        basic={true}
        icon={true}
      >
        <Icon
          as="i"
          name="ellipsis horizontal"
        />
      </Button>
    </div>
  </div>
  <Divider />
</div>
`;
